pkg/cloud/services/networking/router.go

```diff
@@ -165,6 +165,30 @@ INTERFACE_LOOP:
 	return nil
 }
 
+func (s *Service) DeleteRouter(network *infrav1.Network) error {
+	exists, err := s.existsRouter(network.Router.ID)
```

Comment:
I think you still have to check for empty routerID as in you're previous commit. The problem is if the ID is an empty string
and you list all routers in existsRouter without an ID you get all existing routers (same for network)